server/src/main/java/org/jetbrains/bsp/bazel/server/sync/BspProjectMapper.kt

```diff
@@ -289,4 +292,37 @@ class BspProjectMapper(
         val items = modules.mapNotNull(scalaLanguagePlugin::toScalaMainClassesItem)
         return ScalaMainClassesResult(items)
     }
+
+    fun rustWorkspace(
+        project: Project,
+        params: RustWorkspaceParams
+    ): RustWorkspaceResult =
+        collectRustTargets(
+            project,
+            params.targets,
+            languagePluginsService.rustLanguagePlugin::toRustWorkspaceResult
+        )
+
+    fun rustToolchain(
+        project: Project,
+        params: RustToolchainParams
+    ): RustToolchainResult =
+        collectRustTargets(
+            project,
+            params.targets,
+            languagePluginsService.rustLanguagePlugin::toRustToolchains
+        )
+
+    private fun <T> collectRustTargets(
+        project: Project,
+        requestedTargets: List<BuildTargetIdentifier>,
+        resolver: (List<Module>, List<Module>) -> T
```

Comment:
could you name the arguments here?
```suggestion
        resolver: (???: List<Module>, ???: List<Module>) -> T
```

Reply:
Since `RustToolchain` request was deleted, this function was also removed as there was no need for generalization for `RustToolchain` and `RustWorkspace` requests